### DISCUSSION TITLE
How to `delete_all`/`destroy_all` associated records?

### DISCUSSION THREAD

**Discussion by snacks02** [jeremyevans/sequel#1963] [General]
There is `remove_all_*` available, which (as far as I can tell) only removes associations, but not associated records. Is there an easy way to remove associated records as well?

In ActiveRecord, relevant code would be:

```rb
class Book < ApplicationRecord
  has_many :notes, dependent: :delete_all
  # or
  has_many :notes, dependent: :destroy_all
end
```

Or as methods:

```rb
books.delete_all
# or
books.destroy_all
```

If not, would it be the best to just do it manually using the `before_destroy` hook?

**Comment by janko** [comment]
Automatic removal of associations is provided by the [association_dependencies](https://sequel.jeremyevans.net/rdoc-plugins/classes/Sequel/Plugins/AssociationDependencies.html) plugin. The equivalent of Active Record's `delete_all` is the [`delete`](https://sequel.jeremyevans.net/rdoc/classes/Sequel/Dataset.html#method-i-delete) dataset method in Sequel. For `destroy_all`, I think you just have to call `each(&:destroy)` instead.

**Reply by snacks02** [reply]
How would it look like in a model? Like this?

```rb
class Book < Sequel::Model
  # ...
  
  add_association_dependencies notes: :delete
  # or
  add_association_dependencies notes: :destroy
end
```

**Reply by janko** [reply]
Yes, that looks correct to me 👍🏻

**Reply by snacks02** [reply]
Awesome. Thank you!
